api/envoy/extensions/transport_sockets/proxy_protocol/v3/upstream_proxy_protocol.proto

```diff
@@ -17,6 +17,21 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: Upstream Proxy Protocol]
 // [#extension: envoy.transport_sockets.upstream_proxy_protocol]
 
+// Metadata for custom Type-Length-Value (TLV) entries to be sent in the upstream PROXY protocol header.
+message CustomTlvMetadata {
```

Comment:
Will this message ever have another field? If not, I'd say remove it and just have the repeated TlvEntry where needed.

Reply:
SGTM. Removed the type in https://github.com/envoyproxy/envoy/pull/37591/commits/d0a3b622e5727e52cee9117de5dba73b8a57419b.